### ISSUE TITLE
wxDataObjectBase::IsSupported and wxDataObject::IsSupportedFormat

### ISSUE THREAD

**Issue by hwiesmann** [wxWidgets/wxWidgets#24974]
There seem to be a couple of issues with these methods:

1. wxDataObject::IsSupportedFormat is not documented
2. I just checked the macOS and X11 implementations of both classes: they are basically identical (slight difference how the intermediate values are stored)
3. wxDataObject::IsSupportedFormat seems to be declared and defined for each supported operating system.

I suggest removing wxDataObject::IsSupportedFormat or replace its implementation by calling the base class implementation.

**Comment by vadz** [comment]
I agree that it seems weird to have both functions. We probably don't want to remove `IsSupportedFormat()` entirely (somebody is bound to use it), but we can define it just as a trivial inline wrapped calling `IsFormat()` in `wxDataObjectBase` itself.

If you can make the PR doing this, it would be welcome. TIA!
